system/metrics/sinks/prometheus/prometheusReporterService.cpp

```diff
@@ -0,0 +1,354 @@
+/*################################################################################
+#    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+################################################################################
+*/
+#include "prometheusReporterService.hpp"
+
+#include "jsocket.hpp"
+#include "jstream.ipp"
+
+#include "http/platform/httptransport.ipp"
+#include "espcontext.hpp"
+#include "httpsecurecontext.hpp"
+#include "htmlpage.hpp"
+
+using namespace hpccMetrics;
+
+PrometheusReporterService::PrometheusReporterService(const char * sinkname, const IPropertyTree * settings)
+{
+    m_metricsReporter = nullptr;
+    m_metricsSinkName = sinkname;
+    m_processing = false;
+    m_port = DEFAULT_PROMETHEUS_METRICS_SERVICE_PORT;
+    m_use_ssl = DEFAULT_PROMETHEUS_METRICS_SERVICE_SSL;
+    m_sslconfig = nullptr;
+
+    if (settings)
+    {
+        m_port = settings->getPropInt("@port", DEFAULT_PROMETHEUS_METRICS_SERVICE_PORT);
+        m_use_ssl = settings->getPropBool("@useSSL", DEFAULT_PROMETHEUS_METRICS_SERVICE_SSL);
+
+        settings->getProp("@serviceName", m_metricsServiceName);
+        if (m_metricsServiceName.isEmpty())
+            m_metricsServiceName = DEFAULT_PROMETHEUS_METRICS_SERVICE_NAME;
+
+        if(m_use_ssl)
+        {
+               m_sslconfig = settings->getBranch("sslConfig");
+#ifdef _USE_OPENSSL
+            if(m_sslconfig != nullptr)
+                m_ssctx.setown(createSecureSocketContextEx2(m_sslconfig, ServerSocket));
```

Comment:
@afishbeck just realized the createSecureSocketContextEx2 API operates on a "legacy" style config tree and wondered if we should be doing something special to tie into the containerized secrets management. 

sslConfig :
            cipherList:  !el "ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5"
            passphrase: !el someencodedpassphrase
            certificate: !el /home/hpcc/certificate/key.pem
            privatekey: !el /home/hpcc/certificate/public.key.pem
            verify:
            - enable: true
              address_match: true
              accept_selfsigned: true
              trusted_peers: !el a|b|c
              ca_certificates:
              - path: /home/hpcc/certificate

Reply:
Agreed